virttest/qemu_vm.py

```diff
@@ -4464,7 +4494,35 @@ def eject_cdrom(self, device, force=False):
         :param device: device ID;
         :param force: force eject or not;
         """
-        return self.monitor.eject_cdrom(device, force)
+        status_before = status_after = self.check_device_tray_opened(device)
+
+        qmp_monitors = [m for m in self.monitors if m.protocol == "qmp"]
+        for m in qmp_monitors:
+            m.clear_events()
+
+        out = self.monitor.eject_cdrom(device, force)
+
+        for m in qmp_monitors:
+            events = utils_misc.wait_for(m.get_events, timeout=20)
```

Comment:
get_events return [] if there is no event got.
so I think no need line 4507 and 4508

Reply:
utils_misc.wait_for returns None if get_events returns empty list.